### ISSUE TITLE
Don't work hgvs->vcf-variants in some case

### ISSUE THREAD

**Issue by kbaba1001** [chrovis/varity#39]
Conversion is not successful with next mutation at version 0.7.0

```clojure
(require '[varity.hgvs-to-vcf :as h2v]
         '[clj-hgvs.core :as hgvs])

(h2v/hgvs->vcf-variants #clj-hgvs/hgvs "NM_001048171:c.892-2A>G" "path/to/hg38.fa" "path/to/refGene.txt.gz")
;;=> ()
```

However, at version 0.6.2, this case is successful:

```clojure
(h2v/hgvs->vcf-variants #clj-hgvs/hgvs "NM_001048171:c.892-2A>G" "path/to/hg38.fa" "path/to/refGene.txt.gz")
;;=> ({:chr "chr1", :pos 45332046, :ref "T", :alt "C"})
```

I guess that it's the effect of https://github.com/chrovis/varity/pull/37

**Comment by totakke** [comment]
Thank you for the report. I have also confirmed the conversion failure when using the latest `refGene.txt.gz`, but [an old-version `refGene.txt.gz`](https://github.com/chrovis/varity/blob/7f5eb026108c05f4a37270d79a767f17d4e1a848/test/varity/t_common.clj#L37) used in tests works fine.

```clj
(hgvs->vcf-variants #clj-hgvs/hgvs "NM_001048171:c.892-2A>G" "path/to/hg38.fa" "path/to/test-refGene.txt.gz")
;;=> ({:chr "chr1", :pos 45332088, :ref "T", :alt "C"})
```

A change of `NM_001048171` refGene information probably affects the failure. I will further investigate the cause.

**Comment by totakke** [comment]
According to my investigation, CDS end position (`:cds-end`) of NM_001048171 was changed between 2019/10 and 2020/03.

`refGene.txt.gz` in 2019/10:

```clj
(require '[varity.ref-gene :as rg)

(->> (rg/load-ref-genes "path/to/refGene_201910.txt.gz")
                         rg/index
                         (rg/ref-genes "NM_001048171")
                         first
                         (#(select-keys % [:exon-ranges :cds-start :cds-end :strand])))
;;=> {:exon-ranges
;;    [[45329242 45329437]
;;     [45330516 45330557]
;;     ...
;;     [45334391 45334511]
;;     [45340219 45340470]],
;;    :cds-start 45329306,
;;    :cds-end 45340254, ; <-
;;    :strand :reverse}
```

`refGene.txt.gz` in 2020/03:

```clj
(->> (rg/load-ref-genes "path/to/refGene_202003.txt.gz")
                         rg/index
                         (rg/ref-genes "NM_001048171")
                         first
                         (#(select-keys % [:exon-ranges :cds-start :cds-end :strand])))
;;=> {:exon-ranges
;;    [[45329242 45329437]
;;     [45330516 45330557]
;;     ...
;;     [45334391 45334511]
;;     [45340219 45340440]],
;;    :cds-start 45329306,
;;    :cds-end 45334505, ; <-
;;    :strand :reverse}
```

NM_001048171 (MUTYH) is a reverse-strand gene, so the different `:cds-end` derives a different genomic position from the same CDS coordinate (`c.892-2`). In this case, `c.892-2` represents an intron position (`-2` is an offset from an edge of an exon). However, in `refGene.txt.gz` in 2020/03, `c.892-2` is located on an exon. That is wrong representation.

`NM_001048171:c.892-2A>G` variant is found on ClinVar (https://www.ncbi.nlm.nih.gov/clinvar/variation/41766/). A genomic variant is showd as `NC_000001.11:g.45332088T>C` in the page. That corresponds [the conversion result](https://github.com/chrovis/varity/issues/39#issuecomment-804615666) using old `refGene.txt.gz`.

A coding-DNA HGVS depends on a refGene annotation. When the refGene annotation is changed, the HGVS must be updated.

In conclusion, the conversion result at varity 0.7.0 is not failure, or rather the conversion success of varity 0.6.2 is accidental and may lead wrong VCF variants.

**Comment by kbaba1001** [comment]
Thank you!!
